app/api/db/models/factories.py

```diff
@@ -57,19 +58,25 @@ class Meta:
         sqlalchemy_session_persistence = "commit"
 
 
-class EligibilityScreenerFactory(BaseFactory):
+class ExamplePersonFactory(BaseFactory):
```

Comment:
some high level thoughts
1. i think there's a simpler way to generate synthetic data. rather than subclassing SQLAlchemyModelFactory, we should just create a custom Faker provider. here's [an example in the CMS FHIR bid](https://github.com/navapbc/cms-ab2d-bcda-and-dpc/blob/main/api/test/data_generator/fhir_provider.py).
2. synthetic data generation could be test dependency but is ideally not a prod dependency (e.g. Faker should be a dev dependency), so let's move the fake data modules to the `test` folder

Reply:
1. I don't have access to that repo to see the example.
2. You're probably right, kept a similar pattern to what Mass did. I think a few non-local only scripts did rely on the factory, but shouldn't have.

Reply:
1. weird, i'll try to get you access, in the meantime feel free to consider this discussion non-blocking (can address in a future PR)

Reply:
Moved the factory to the test folders + moved its dependencies to be dev only.

Reply:
@chouinar you should have access now to the bid repo, try again

Reply:
I think there's two aspects to data generation at play. General data generation, and populating a test DB.

I do like the approach you linked regarding generating data regardless of the context. Might be worth a follow-up to add something similar.

What would be your suggestion for populating a DB with test scenarios? Assuming we use an ORM, the factory approach does help with setting up several tables at once via the relationships. Without the ORM though, I'm curious how you would structure that? Ideally not needing to manually insert all the records into the DB (which I've done on projects before which just leads to developers not testing the complex cases).

Reply:
> What would be your suggestion for populating a DB with test scenarios? Assuming we use an ORM, the factory approach does help with setting up several tables at once via the relationships. Without the ORM though, I'm curious how you would structure that? Ideally not needing to manually insert all the records into the DB (which I've done on projects before which just leads to developers not testing the complex cases).

we might need to dig deeper into some examples to figure this out, since i think there are different testing situations that would merit different approaches.

i think there's the low level testing of the repository/persistence layer itself (no business logic, just testing that the queries work, that low level inserts, updates, and deletes work and don't leave the database in an inconsistent state). for this we need to populate the DB with varying fixtures to test out the query logic, and i agree with you that some sort of factory pattern could work well for that.

for testing out the more complex services layer that has the business logic, that topic probably merits more discussion with some examples.

Reply:
Added https://github.com/navapbc/template-application-flask/issues/7 as a general DB/data follow-up